airbyte-integrations/connectors/source-gitlab/source_gitlab/source.py

```diff
@@ -69,7 +71,17 @@ def get_new_token_expiry_date(access_token_expires_in: int, access_token_created
         return pendulum.from_timestamp(access_token_created_at + access_token_expires_in)
 
     def refresh_access_token(self) -> Tuple[str, int, int, str]:
-        response_json = self._get_refresh_access_token_response()
+        try:
```

Comment:
can this be implemented in the Airbyte CDK?

Reply:
I agree, that the `400` should be handled on the CDK level. However, some Data Providers could get the same error using a different `status_code`, like `500` or `401` in some cases. + I don't think we should include the error message in the CDK implementation.

Reply:
We can override status codes and messages on the connector level, what do you think?